claat/proto-renderer/generic_renderer_test.go

```diff
@@ -0,0 +1,59 @@
+package genrenderer
+
+import (
+	"go/build"
+	"path/filepath"
+	"testing"
+	"text/template"
+)
+
+type encapsulatedTest struct {
+	In      sampleProtoTemplate
+	Out     string
+	WantErr bool
+}
+
+type sampleProtoTemplate struct {
+	Value interface{}
+}
+
+func newSampleProtoTemplate(el interface{}) sampleProtoTemplate {
+	return sampleProtoTemplate{Value: el}
+}
+
+func runEncapsulatedTest(test encapsulatedTest, tmpl *template.Template, t *testing.T) {
```

Comment:
Need to encapsulate the scope of possible panics, else tests won't be executed past the first panic on a loop.

Reply:
Have you checked to make sure this actually works? I have a suspicion there might be scoping issues associated with performing test ops in deferred code.

Reply:
what do you mean by "actually works"? the tests pass and flipping the `WantErr` booleans cause tests to break - as expected.

Reply:
update: verified having an inline func instead of a helper still works. helper functions seem fine or anything needs to change? at least their order within the file?

Reply:
Inline is fine.